src/defrag.c

```diff
@@ -276,9 +276,9 @@ void activeDefragZsetEntry(zset *zs, dictEntry *de) {
     double *newscore;
     sds sdsele = dictGetKey(de);
     if ((newsds = activeDefragSds(sdsele))) dictSetKey(zs->dict, de, newsds);
-    newscore = zslDefrag(zs->zsl, *(double *)dictGetVal(de), sdsele, newsds);
+    newscore = zslDefrag(zs->zsl, dictGetDoubleVal(de), sdsele, newsds);
     if (newscore) {
-        dictSetVal(zs->dict, de, newscore);
+        dictSetDoubleVal(de, *newscore);
     }
```

Comment:
We no longer need to do this defragmentation in your proposal, since the value at newscore will be the same as before. You should be able to delete this whole if statement and change the return type of zslDefrag.

Reply:
Yeah, You're right, I've fixed that as you suggested.